models/mli/model_decision_tree_linear_combo.py

```diff
@@ -0,0 +1,288 @@
+"""Decision tree plus linear model"""
+
+
+import uuid
+import os
+import datatable as dt
+import numpy as np
+from h2oaicore.models import CustomModel
+from sklearn.preprocessing import LabelEncoder
+from h2oaicore.systemutils import physical_cores_count
+from h2oaicore.systemutils import temporary_files_path, remove, config
+from h2oaicore.systemutils import make_experiment_logger, loggerinfo, loggerwarning, loggerdebug
+
+
+class DECISION_TREE_PLUS_LINEAR(CustomModel):
+    _regression = True
+    _binary = True
+    _multiclass = False
+    _display_name = "DECISION_TREE_PLUS_LINEAR"
+    _description = "Takes the results of a decision tree and then fits a linear model to each set of node data"
+    _modules_needed_by_name = ['shap', 'collections', 'scipy', 'pandas', 
+                               'matplotlib', 'sklearn']
+
+    @staticmethod
+    def do_acceptance_test():
+        return True
+
+    def set_default_params(self, accuracy=None, time_tolerance=None,
+                           interpretability=None, **kwargs):
+        # Fill up parameters we care about
+        self.params = dict(random_state=kwargs.get("random_state", 1234),
+                           max_depth_duplication = None, n_estimators= 10,
+                           tree_depth=3)
+
+    def mutate_params(self, accuracy=10, **kwargs):
+        if accuracy > 8:
+            tree_depth = [4, 5]
+        elif accuracy >= 5:
+            tree_depth = [3, 4]
+        else:
+            tree_depth = [2]
+
+        self.params["tree_depth"] = np.random.choice(tree_depth) 
```

Comment:
Is there reasoning behind parameters selected for these models? Maybe good to comment.